src/sage/structure/coerce_actions.pyx

```diff
@@ -224,15 +241,15 @@ def detect_element_action(Parent X, Y, bint X_on_left, X_el=None, Y_el=None):
         except CoercionException as msg:
             _record_exception()
 
-    # element x defining _act_on_
-    try:
-        if x._act_on_(y, X_on_left) is not None:
-            return ActOnAction(X, Y, X_on_left, False)
-    except CoercionException:
-        _record_exception()
-
-    # element x defining _acted_upon_
     if isinstance(Y, Parent):
+        # element y defining _act_on_
+        try:
+            if y._act_on_(x, not X_on_left) is not None:
```

Comment:
If we follow the docstring `as defined by elements of X`, this looks wrong.  Indeed, in the old code, we were only looking at methods of `x`.

Reply:
I don't quite think it is saying that it defined by the methods of `x` (the action is defined by `x` using the method of `y` is how I read this), but you can read it that way. Although I don't think the docstring is accurate in this regard either.